### PR TITLE
fix(cmf/httpMiddleware): override default headers

### PR DESCRIPTION
**Pull request by jlame**
**What is the problem this PR is trying to solve?**
Content-type is always sent which causes some issue with file upload

**What is the chosen solution to this problem?**
Removing the content type from header when a formdata is sent (like in sagas/http)

**Please check if the PR fulfills these requirements**

* [ ] The PR commit message follows our [guidelines](https://github.com/talend/tools/blob/master/tools-root-github/CONTRIBUTING.md)
* [ ] Tests for the changes have been added (for bug fixes / features) And [non reg](./screenshots.md) done before need review
* [ ] Docs have been added / updated (for bug fixes / features)
* [ ] Related design / discussions / pages (not in jira), if any, are all linked or available in the PR

**[ ] This PR introduces a breaking change**

### CODE REVIEW

File: packages/cmf/src/middlewares/http/middleware.js [hunk 1]
```diff
@@ -101,10 +101,16 @@ export function getMethod(action) {
...
 export function mergeConfiguredHeader(config) {
 	// still need to keep the previous header added by action
-	return options => ({
-		...options,
-		headers: { ...options.headers, ...DEFAULT_HTTP_HEADERS, ...config.headers },
-	});
+	return options => {
+		const headerMergedConfig = {
+			...options,
+			headers: { ...DEFAULT_HTTP_HEADERS, ...options.headers, ...config.headers },
```
Comment (romainseb) [inline packages/cmf/src/middlewares/http/middleware.js]: Shouldn't be more like 
`{ ...DEFAULT_HTTP_HEADERS, ....config.headers, ...options.headers }`
to allow specific options to override global config ?

Comment (jlame) [reply packages/cmf/src/middlewares/http/middleware.js]: you're right about that options are the one coming from the action so it should be the last one overriding

File: packages/cmf/src/middlewares/http/middleware.js [hunk 2]
```diff
@@ -101,10 +101,16 @@ export function getMethod(action) {
...
-	return options => ({
-		...options,
-		headers: { ...options.headers, ...DEFAULT_HTTP_HEADERS, ...config.headers },
-	});
+	return options => {
+		const headerMergedConfig = {
+			...options,
+			headers: { ...DEFAULT_HTTP_HEADERS, ...config.headers, ...options.headers },
+		};
+		if (headerMergedConfig.body && headerMergedConfig.body instanceof FormData) {
```
Comment (jmfrancois) [inline packages/cmf/src/middlewares/http/middleware.js]: why the first condition is required ? If no .body it will not be an instance of FormData right.

Comment (jlame) [reply packages/cmf/src/middlewares/http/middleware.js]: right changing

**Pull request merged**